src/Module/AbstractBaseModule.php

```diff
@@ -0,0 +1,448 @@
+     */
+    use NormalizeStringCapableTrait;
+
+    /*
+     * Provides iterable normalization functionality.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeIterableCapableTrait;
+
+    /*
+     * Provides container normalization functionality.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeContainerCapableTrait;
+
+    /*
+     * Provides common functionality for events.
+     *
+     * @since [*next-version*]
+     */
+    use EventsFunctionalityTrait;
+
+    /*
+     * Provides functionality for creating invalid-argument exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /*
+     * Provides functionality for creating out-of-range exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateOutOfRangeExceptionCapableTrait;
+
+    /*
+     * Provides functionality for creating internal exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateInternalExceptionCapableTrait;
+
+    /*
+     * Provides functionality for creating runtime exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateRuntimeExceptionCapableTrait;
+
+    /*
+     * Provides functionality for creating container exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateContainerExceptionCapableTrait;
+
+    /*
+     * Provides functionality for creating container not-found exceptions.
+     *
+     * @since [*next-version*]
+     */
+    use CreateNotFoundExceptionCapableTrait;
+
+    /*
+     * Provides string translating functionality.
+     *
+     * @since [*next-version*]
+     */
+    use StringTranslatingTrait;
+
+    /**
+     * The factory to use for creating containers.
+     *
+     * @since [*next-version*]
+     *
+     * @var ContainerFactoryInterface
+     */
+    protected $containerFactory;
+
+    /**
+     * The factory to use for creating config containers.
+     *
+     * @since [*next-version*]
+     *
+     * @var ContainerFactoryInterface
+     */
+    protected $configFactory;
+
+    /**
+     * The factory to use for creating composite containers.
+     *
+     * @since [*next-version*]
+     *
+     * @var ContainerFactoryInterface
+     */
+    protected $compContainerFactory;
+
+    /**
+     * Initializes the module with all required information.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable         $key                  The module key.
+     * @param string[]|Stringable[]     $dependencies         The module dependencies.
+     * @param ContainerFactoryInterface $containerFactory     The container factory.
+     * @param ContainerFactoryInterface $configFactory        The config factory.
+     * @param ContainerFactoryInterface $compContainerFactory The composite container factory.
+     */
+    protected function _initModule(
+        $key,
+        $dependencies = [],
+        ContainerFactoryInterface $configFactory,
```

Comment:
This should be [`ConfigFactoryInterface`][]. See also type docs and other code below.

[`ConfigFactoryInterface`]: https://github.com/Dhii/config-interface/blob/develop/src/ConfigFactoryInterface.php